MomenTone/app/src/main/java/lighterletter/c4q/nyc/momentone/DrawingView.java

```diff
@@ -256,4 +258,35 @@ public void startNew() {
         drawCanvas.drawColor(0, PorterDuff.Mode.CLEAR);
         invalidate(); //clears the canvas and updates the display.
     }
+
+    /*
+     * Overriding the custom view's save instance state and restore state to capture
+     * the values that are needed for orientation changes.
+     * If the values should persist after app is closed, then you can save the values in
+     * shared preferences. Not sure if that is what you wanted.
+     */
+    @Override
+    protected Parcelable onSaveInstanceState() {
+        Bundle bundle = new Bundle();
+        bundle.putParcelable("instanceState", super.onSaveInstanceState());
+//        bundle.putParcelable("drawCanvas", (Parcelable) this.drawCanvas);
+        bundle.putParcelable("canvasBitmap", this.canvasBitmap);
```

Comment:
as mentioned in other lectures, persisting a bitmap is often times overkill and can result in poor performance.  consider saving the state of the bitmap (points, paths, colors, stroke widths) instead and restoring the canvas state, as needed.